libtalek/client_test.go

```diff
@@ -126,3 +126,42 @@ func TestRead(t *testing.T) {
 		t.Fatalf("Read wasn't for the enqueued subscription. %v / %v / %d", rv1, rv2, bucket)
 	}
 }
+
+func TestFragment(t *testing.T) {
+	theMsg := make([]byte, 1024*256)
+	validateMsg := make([]byte, 1024*256)
+
+	hdr := msgHeader{}
+	hdr.left = uint32(len(theMsg))
+
+	parts := make([][]byte, 0)
+	for hdr.left > 0 {
```

Comment:
This seems like a really odd abstraction. the msgHeader.Fill is something that the caller needs to iteratively allocate, fill, and advance the message. 

I think what I had in mind, was a Message class. Where a Marshal method would just spit out a array of partial messages and the Unmarshal method would take in all the partials and recompose the message. 

This method seems similarly error prone.
For example, you don't have an ID for the message. If you've missed some messages, you don't know if you're still unpacking the same message or a different one. I'm expecting that we have tests for all of these different failure scenarios. 

When we didn't do multi-message, all these failure scenarios were implicitly covered because messages were self-contained. Now that you're splitting them apart, I'd expect robustness against message drops.